### ISSUE TITLE
Setting window icon in Avalonia on a Windows system doesn't seem to work with AOT

### ISSUE THREAD

**Issue by IronEaterBeast** [AvaloniaUI/Avalonia#13824]
Both of the following methods don't work, and the program won't start.

`this.Icon = bitmap;`

`Icon="/Assets/icontest.ico"`

But in a non-AOT scenario, the program works just fine.

**Comment by timunie** [comment]
@IronEaterBeast yeah but in this case you need to make sure things you need are not trimmed. That's what I wanted to say.

**Comment by timunie** [comment]
Check docs at micrsoft / dotNET please. I'm struggling with it also often. What I can say is try to avoid reflection, add a roots.xml if needed and use CompiledBindings anywhere.

**Comment by SweetLoverFT** [comment]
Let me answer this question:
1. Icon is WindowIcon type while Bitmap is another type, so they're different
2. You should use compiled binding MVVM as default
```csharp
public WindowIcon? LoadIcon(string iconName)
{
    try
    {
        var uri = new Uri($"{_prefix}Icons/{iconName}");
        return new WindowIcon(new Bitmap(AssetLoader.Open(uri)));
    }
    catch (Exception ex)
    {
        return null;
    }
}
```
The `prefix` is `_prefix = $"avares://{AppDomain.CurrentDomain.FriendlyName}/Assets/";`
